examples/DEVELOPER.md

```diff
@@ -0,0 +1,210 @@
+# Adding a sandbox example
+
+## Add a `verify.sh` test to your sandbox
+
+Sandboxes are tested as part of the continuous integration process, which expects
+each sandbox to have a `verify.sh` script containing tests for the example.
+
+### Basic layout of the `verify.sh` script
+
+At a minimum the `verify.sh` script should include the necessary parts to start
+and stop your sandbox.
+
+Given a sandbox with a single `docker` composition, adding the following
+to `verify.sh` will test that the sandbox can be started and stopped.
+
+```bash
+#!/bin/bash -e
+
+export NAME=example-sandbox
+
+# shellcheck source=examples/verify-common.sh
+. "$(dirname "${BASH_SOURCE[0]}")/../verify-common.sh"
+
+# add example tests here...
+
+```
+
+The `$NAME` variable is used for logging when testing the example, and will
+often be the same as the directory name.
+
+### Log before running each test
+
+There is a utility function `run_log` that can be used to indicate in test logs what
+is being executed and why, for example:
+
+```bash
+run_log "Checking foo.txt was created"
+ls foo.txt
+
+run_log "Checking bar.txt was created"
+ls bar.txt
+```
+
+### Add tests reflecting the documented examples
+
+The tests should follow the steps laid out in the documentation.
+
+For example, if the documentation provides a series of `bash` commands to execute, add these in order to `verify.sh`.
+
+You may wish to grep the responses, or check return codes to ensure the commands respond as expected.
+
+Likewise, if the documentation asks the user to browse to a page - for example http://localhost:8000 -
+then you should add a test to ensure that the given URL responds as expected.
+
+If an example web page is also expected to make further Javascript `HTTP` requests in order to function, then add
```

Comment:
Not sure if you want to follow up fixing this: s/Javascript/JavaScript https://developer.mozilla.org/en-US/docs/Web/JavaScript 😅